From ed2568dd3fe6b5db15eebcfc723d9c110b67d670 Mon Sep 17 00:00:00 2001
From: John Wiele <jwiele@redhat.com>
Date: Fri, 4 Sep 2020 09:55:56 -0400
Subject: [PATCH 1/2] Updated vdoformat to build with older versions of blkid.

Author: jwiele@redhat.com
Author: muhammad.ahmad@seagate.com

Enable compilation on CentOS 7.*.

Older versions of blkid, such as version 2.3 as found on
CentOS 7.8, may not define BLKID_SUBLKS_BADCSUM. See pull
request: https://github.com/dm-vdo/vdo/pull/34.

.../vdoFormat.c
Fixed to only include BLKID_SUBLKS_BADCSUM in superblock probe
  flags if it is defined.
Fixed white space that got messed up by a previous change.
---
 utils/vdo/user/vdoformat.c | 15 ++++++++-------
 1 file changed, 8 insertions(+), 7 deletions(-)

diff --git a/utils/vdo/user/vdoformat.c b/utils/vdo/user/vdoformat.c
index f3192e82..9ec533ac 100644
--- a/utils/vdo/user/vdoformat.c
+++ b/utils/vdo/user/vdoformat.c
@@ -289,13 +289,14 @@ static int checkForSignaturesUsingBlkid(const char *filename, bool force)
   blkid_probe_set_partitions_flags(probe, BLKID_PARTS_MAGIC);
 
   blkid_probe_enable_superblocks(probe, 1);
-  blkid_probe_set_superblocks_flags(probe, BLKID_SUBLKS_LABEL |
-				    BLKID_SUBLKS_UUID |
-				    BLKID_SUBLKS_TYPE |
-				    BLKID_SUBLKS_USAGE |
-				    BLKID_SUBLKS_VERSION |
-				    BLKID_SUBLKS_MAGIC |
-				    BLKID_SUBLKS_BADCSUM);
+  int flags = BLKID_SUBLKS_DEFAULT | BLKID_SUBLKS_USAGE | BLKID_SUBLKS_VERSION
+    | BLKID_SUBLKS_MAGIC;
+
+#ifdef BLKID_SUBLKS_BADCSUM
+  flags |= BLKID_SUBLKS_BADCSUM;
+#endif
+
+  blkid_probe_set_superblocks_flags(probe, flags);
 
   Buffer *buffer = NULL;
   result = makeBuffer(0, &buffer);
-- 
2.35.3

